### ISSUE TITLE
Pokémon HeartGold with "Refined Gold" patch crashes with red screens and data abort error

### ISSUE THREAD

**Issue by Hardback247** [DS-Homebrew/nds-bootstrap#1528]
**Your console:**
New 3DS XL

**Launch Method:**
SD card

**Version used:**
TwiLight Menu++ v25.8.1 and nds-bootstrap v0.70.0

**Game tested/used/applicable:**
Pokémon HeartGold Version, with the "Refined Gold" ROM hack patched to it.
Homebrewed

**Save file:**

**Expected Behavior:**
It should work normally.

**Actual Behavior:**
It crashes to red "Error: Data Abort!" screens suddenly after playing for a while, like sometimes when a battle starts, or after exiting or entering a building.

### Steps to reproduce
1. Start a new game, and either play for a certain amount of time, or just go in and out of the character's house for a few minutes.
2. Red screen appears showing a "Data Abort" error

### nds-bootstrap.ini
[NDS-BOOTSTRAP]
NDS_PATH = sd:/3ds/4780 - Pokemon HeartGold (U)(Xenophobia) (patched).nds
SAV_PATH = sd:/3ds/saves/4780 - Pokemon HeartGold (U)(Xenophobia) (patched).sav
BOOST_CPU = 0
DEBUG = 0
LOCK_ARM9_SCFG_EXT = 0
CACHE_FAT_TABLE = 0
AP_FIX_PATH =
BOOST_VRAM = 0
CARD_READ_DMA = 1
ASYNC_CARD_READ = 0
DSI_MODE = 1
SWI_HALT_HOOK = 1
EXTENDED_MEMORY = 0
DONOR_SDK_VER = 0
PATCH_MPU_REGION = 0
PATCH_MPU_SIZE = 0
CONSOLE_MODEL = 2
LANGUAGE = -1
REGION = 1
LOGGING = 0
B4DS_MODE = 0
ROMREAD_LED = 0
DMA_ROMREAD_LED = -1
PRECISE_VOLUME_CONTROL = 0
SDNAND = 0
MACRO_MODE = 0
SLEEP_MODE = 1
SOUND_FREQ = 0
HOTKEY = 284
USE_ROM_REGION = 1
RAM_DRIVE_PATH = sd:/null.img
GUI_LANGUAGE = en
FORCE_SLEEP_PATCH = 0

![IMG_3747](https://user-images.githubusercontent.com/125831518/220179355-05acfde3-d13a-4e01-a167-bc64c9079bba.JPG)

### Other notes
 I've tried FOUR different HeartGold ROMs, but they all crashed with the red screens.

"4780 - Pokemon HeartGold(U)(Xenophobia)"
"Pokemon HeartGold Version"
"Pokemon - HeartGold Version (USA)"
"Pokemon Heart Gold"

### Will you be available for future issues and/or replies?
Yes

**Comment by Hardback247** [comment]
> Have you tried AP-patching the clean ROM before applying the ROM hack patch?

How do you do that?

**Comment by RocketRobz** [comment]
> > Have you tried AP-patching the clean ROM before applying the ROM hack patch?
> 
> How do I do that?

Use DS-Scene ROM Tool to patch the clean ROM.

**Comment by Hardback247** [comment]
> > > Have you tried AP-patching the clean ROM before applying the ROM hack patch?
> > 
> > 
> > How do I do that?
> 
> Use DS-Scene ROM Tool to patch the clean ROM.

I don't know where to download it. On the gbatemp page, it says that "permission must be given prior to redistributing [their] files/updates. Offenders will be reported and links removed."

**Comment by RocketRobz** [comment]
> I don't know where to download it. On the gbatemp page, it says that "permission must be given prior to redistributing [their] files/updates. Offenders will be reported and links removed."

That just means that reuploading isn't allowed.

**Comment by Hardback247** [comment]
> > I don't know where to download it. On the gbatemp page, it says that "permission must be given prior to redistributing [their] files/updates. Offenders will be reported and links removed."
> 
> That just means that reuploading isn't allowed. There should be an orange button to download it.

Oh, I see it. Thanks.

The program says that the vanilla ROM of HeartGold I have is validated. What do I do now?

**Comment by RocketRobz** [comment]
> Oh, I see it. Thanks.
> 
> The program says that the vanilla ROM of HeartGold I have is validated. What do I do now?

The AP-patch should automatically be enabled, so just export the ROM.

**Comment by Hardback247** [comment]
Thanks! It works now. I do have another question about two other patched ROMs for Black and Black 2. Should I start a new "issue"?

**Comment by RocketRobz** [comment]
If the same AP-patching process doesn't fix them, then yes.

**Comment by mojius** [comment]
Sorry to bump after a year, but the same issue is happening to me except I'm using ntr-forwarder. Did everything you said above -- AP patch, then xdelta. I added my own usrcheat.dat with a new code and enabled that for a while, then disabled when I didn't need it. Still random crashes. Please help.

**Comment by RocketRobz** [comment]
> Sorry to bump after a year, but the same issue is happening to me except I'm using ntr-forwarder. Did everything you said above -- AP patch, then xdelta. I added my own usrcheat.dat with a new code and enabled that for a while, then disabled when I didn't need it. Still random crashes. Please help.

Which Refined Gold are you playing? You shouldn't need to AP-patch the ROM.

**Comment by mojius** [comment]
Your reply speed is unbelievable! :O Thank you so much!
I'm playing Refined Gold 3.2, but this happened with a non-AP patched hack as well.

**Comment by RocketRobz** [comment]
> Your reply speed is unbelievable! :O Thank you so much! I'm playing Refined Gold 3.2, but this happened with a non-AP patched hack as well. Was also trying to use a widescreen hack, so the TWL sysfirm(?) is definitely widescreen patched, but I doubt _that_ would cause the bootstrap/forwarder to bug out.

If what you're playing is Refined Gold Overhauled, then you'll need to update the ROM hack.

**Comment by mojius** [comment]
> > Your reply speed is unbelievable! :O Thank you so much! I'm playing Refined Gold 3.2, but this happened with a non-AP patched hack as well. Was also trying to use a widescreen hack, so the TWL sysfirm(?) is definitely widescreen patched, but I doubt _that_ would cause the bootstrap/forwarder to bug out.
> 
> If what you're playing is Refined Gold Overhauled, then you'll need to update the ROM hack.

No, just the regular one.

**Comment by RocketRobz** [comment]
> > > Your reply speed is unbelievable! :O Thank you so much! I'm playing Refined Gold 3.2, but this happened with a non-AP patched hack as well. Was also trying to use a widescreen hack, so the TWL sysfirm(?) is definitely widescreen patched, but I doubt _that_ would cause the bootstrap/forwarder to bug out.
> > 
> > 
> > If what you're playing is Refined Gold Overhauled, then you'll need to update the ROM hack.
> 
> No, just the regular one.

Seeing as the latest version of Refined Gold is v3.2.1, you'll need to update anyways.

**Comment by mojius** [comment]
Yeah, that's the one I have. When you go to download 3.2.1, you get a 3.2 patch in the zip.

**Comment by mojius** [comment]
Thank you so much for your help -- I didn't update because I'm on the newest version. What else could it be? Should I open a new ticket?

**Comment by RocketRobz** [comment]
> Thank you so much for your help -- I didn't update because I'm on the newest version. What else could it be? Should I open a new ticket?

When did you create the forwarder? Also, does `apfix.pck` exist in `sd:/_nds/ntr-forwarder/`?

**Comment by mojius** [comment]
Yes it does. I created and downloaded the forwarder a few days ago.

**Comment by mojius** [comment]
Going to delete everything in _nds for a full reinstall.

**Comment by mojius** [comment]
Full reinstall, still happens, even when running through Twilight Menu.

**Comment by optronix10** [comment]
I too am experiencing the same red screen error when using the NDS Forwarder to launch the refined gold hack. I also have the apfix.pck file in the appropriate folder. 

Any ideas on how to remedy this?

**Comment by RocketRobz** [comment]
> I too am experiencing the same red screen error when using the NDS Forwarder to launch the refined gold hack. I also have the apfix.pck file in the appropriate folder.
> 
> Any ideas on how to remedy this?

Which version of nds-bootstrap are you using? You can check by pressing L+Down+Select while running the game.

**Comment by optronix10** [comment]
v2.4.2

Thanks for the quick reply!

**Comment by RocketRobz** [comment]
> v2.4.2
> 
> Thanks for the quick reply!

Please update nds-bootstrap then.

**Comment by optronix10** [comment]
I've replaced the old nds-boostrap w/ v2.4.4 . I will report back if I encounter the error again.

**Comment by optronix10** [comment]
> > v2.4.2
> > Thanks for the quick reply!
> 
> Please update nds-bootstrap then.

Just ran into the Error: Data Abort! again after updating to 2.4.4

**Comment by RocketRobz** [comment]
> Just ran into the Error: Data Abort! again after updating to 2.4.4

When does the error occur after launching the game?

**Comment by optronix10** [comment]
it seems to typically occur after I have been playing for over 10 mins and am entering or exiting an area.

**Comment by RocketRobz** [comment]
> it seems to typically occur after I have been playing for over 10 mins and am entering or exiting an area.

Which version of the ROM hack are you playing?

**Comment by optronix10** [comment]
V4.1.1

**Comment by optronix10** [comment]
This was the method I used to increase the games fps.

https://www.reddit.com/r/Delta_Emulator/s/KVL9m0N4tv

**Comment by RocketRobz** [comment]
> V4.1.1
> 
> I am not sure if this could contribute, but I briefly used a hex editor on the ROM to allow it to run closer to 60fps, but once I saw the red screen error, I reinstalled the backup/pre-hex edited version. I did keep using the same save file though.

I thought the latest is v4.0.4?
Where did you get v4.1.1 from?

**Comment by optronix10** [comment]
From the link on Pokemon community. See below.

https://www.pokecommunity.com/threads/refined-gold-overhaul-complete.534484/

**Comment by optronix10** [comment]
> > V4.1.1
> > I am not sure if this could contribute, but I briefly used a hex editor on the ROM to allow it to run closer to 60fps, but once I saw the red screen error, I reinstalled the backup/pre-hex edited version. I did keep using the same save file though.
> 
> I thought the latest is v4.0.4? Where did you get v4.1.1 from?

Do you think the version could be causing the problem? From the comments, it seems like they may be releasing another version in the next few days.

**Comment by RocketRobz** [comment]
> From the link on Pokemon community. See below.
> 
> https://www.pokecommunity.com/threads/refined-gold-overhaul-complete.534484/

Ah, I was not aware a new version released. I'd have to add an AP-fix for that version in order for it to not crash.

**Comment by optronix10** [comment]
> > From the link on Pokemon community. See below.
> > https://www.pokecommunity.com/threads/refined-gold-overhaul-complete.534484/
> 
> Ah, I was not aware a new version released. I'd have to add an AP-fix for that version in order for it to not crash.

That makes sense. It looks like they just released 4.1.2.

Are you planning to add an updated AP-fix for this hack? Unfortunately, I am unsure how much time and effort this would require. Thank you for all that you have done thus far and being so quick to reply to my questions!

**Comment by qwertyey** [comment]
Has anything been done regarding the AP-fix for this romhack? I have reached a point in my playthrough where I am unable to pass an area without the red screen.

**Comment by qwertyey** [comment]
> > Has anything been done regarding the AP-fix for this romhack? I have reached a point in my playthrough where I am unable to pass an area without the red screen.
> 
> Make sure the ROM hack, TWLMenu++, and nds-bootstrap are on their latest versions.

nds-bootstrap is on v2.4.6 and the ROM hack is on v4.1.2. I redscreen on trying to enter the National Park as well as exiting a Pokecenter. One of my concerns is the difference in the patched ROM's size (121.8MB) vs the base ROM (128MB).

**Comment by RocketRobz** [comment]
> > > Has anything been done regarding the AP-fix for this romhack? I have reached a point in my playthrough where I am unable to pass an area without the red screen.
> > 
> > 
> > Make sure the ROM hack, TWLMenu++, and nds-bootstrap are on their latest versions.
> 
> nds-bootstrap is on v2.4.6 and the ROM hack is on v4.1.2. I redscreen on trying to enter the National Park as well as exiting a Pokecenter. One of my concerns is the difference in the patched ROM's size (121.8MB) vs the base ROM (128MB).

If the red screen crash only occurs for those areas, then the ROM hack might be bugged.

**Comment by qwertyey** [comment]
> If the red screen crash only occurs for those areas, then the ROM hack might be bugged.

It happens frequently, particularly when entering or exiting random areas – and occasionally at random times. Sometimes the game simply freezes up on an event such as Pokeballs spinning infinitely in trainer battle intros or initiating a menu (selecting Pokemon or opening bag). These two areas just became consistent crashes only after defeating Whitney. I am happy to provide anything that might help.
